src/vms/pvm/etna-builder/spendHelper.ts

```diff
@@ -0,0 +1,333 @@
+   * @returns {boolean} - Returns true if the asset should be consumed, false otherwise.
+   */
+  shouldConsumeLockedAsset(assetId: string): boolean {
+    return this.toStake.has(assetId) && this.toStake.get(assetId) !== 0n;
+  }
+
+  /**
+   * Determines if an asset should be consumed based on its asset ID.
+   *
+   * @param {string} assetId - The ID of the asset to check.
+   * @returns {boolean} - Returns true if the asset should be consumed, false otherwise.
+   */
+  shouldConsumeAsset(assetId: string): boolean {
+    return (
+      (this.toBurn.has(assetId) && this.toBurn.get(assetId) !== 0n) ||
+      this.shouldConsumeLockedAsset(assetId)
+    );
+  }
+
+  /**
+   * Consumes a locked asset based on its asset ID and amount.
+   *
+   * @param {string} assetId - The ID of the asset to consume.
+   * @param {bigint} amount - The amount of the asset to consume.
+   * @returns {bigint} The remaining amount of the asset after consumption.
+   */
+  consumeLockedAsset(assetId: string, amount: bigint): bigint {
+    if (amount < 0n) {
+      throw new Error('Amount to consume must be greater than or equal to 0');
+    }
+
+    const remainingAmountToStake = this.toStake.get(assetId) ?? 0n;
+
+    // Stake any value that should be staked
+    const amountToStake = bigIntMin(
+      // Amount we still need to stake
+      remainingAmountToStake,
+      // Amount available to stake
+      amount,
+    );
+
+    this.toStake.set(assetId, remainingAmountToStake - amountToStake);
+
+    return amount - amountToStake;
+  }
+
+  /**
+   * Consumes an asset based on its asset ID and amount.
+   *
+   * @param {string} assetId - The ID of the asset to consume.
+   * @param {bigint} amount - The amount of the asset to consume.
+   * @returns {bigint} The remaining amount of the asset after consumption.
+   */
+  consumeAsset(assetId: string, amount: bigint): bigint {
+    if (amount < 0n) {
+      throw new Error('Amount to consume must be greater than or equal to 0');
+    }
+
+    const remainingAmountToBurn = this.toBurn.get(assetId) ?? 0n;
+
+    // Burn any value that should be burned
+    const amountToBurn = bigIntMin(
+      // Amount we still need to burn
+      remainingAmountToBurn,
+      // Amount available to burn
+      amount,
+    );
+
+    this.toBurn.set(assetId, remainingAmountToBurn - amountToBurn);
+
+    // Stake any remaining value that should be staked
+    return this.consumeLockedAsset(assetId, amount - amountToBurn);
+  }
+
+  /**
+   * Calculates the fee for the SpendHelper based on its complexity and gas price.
+   *
+   * @returns {bigint} The fee for the SpendHelper.
+   */
+  calculateFee(): bigint {
+    this.consolidateOutputs();
+
+    const gas = dimensionsToGas(this.getComplexity(), this.weights);
+
+    return gas * this.gasPrice;
+  }
+
+  calculateFeeWithTemporaryOutputComplexity(
+    transferableOutput: TransferableOutput,
+  ): bigint {
+    const oldOutputComplexity = this.outputComplexity;
```

Comment:
Why does this function live on the `spendHelper`? If this is the only function that writes the outputComplexity -- and then reverts it -- can we get rid of that state and just use static functions?

Reply:
Removed and refactored.